### PR TITLE
TN-615 allow empty study id && site id

### PR DESCRIPTION
**Pull request by achen2401**
address TN-615:
allow study id and site id to be changed back to empty value:

https://jira.movember.com/browse/TN-615

### CODE REVIEW

File: portal/static/js/main.js [hunk 1]
```diff
@@ -1894,30 +1894,37 @@ var assembleContent = {
                 };
             };
 
-            var studyId = $("#profileStudyId").val();
-            var siteId = $("#profileSiteId").val();
+            var studyIdField = $("#profileStudyId");
+            var siteIdField = $("#profileSiteId");
+            var hasStudyId = studyIdField.length > 0 && studyIdField.is(":visible");
```
Comment (ivan-c) [inline portal/static/js/main.js]: Is the visibility of these fields triggered by something else you should be checking the state of instead?

File: portal/static/js/main.js [hunk 2]
```diff
@@ -1894,30 +1894,37 @@ var assembleContent = {
...
+                                identifier.system != SYSTEM_IDENTIFIER_ENUM["practice_region"]) {
+                                identifiers.push(identifier);
+                            }
                         });
-                    };
-                }).fail(function() {
-                   // console.log("Problem retrieving data from server.");
+                    }
+                }).fail(function(xhr) {
+                   tnthAjax.reportError(userId, "api/demographics"+userId, xhr.responseText);
```
Comment (ivan-c) [inline portal/static/js/main.js]: Does `tnthAjax.reportError` log to the frontend console as well? I can see error reporting being useful in both places :)

File: portal/static/js/main.js (hunk 1 above)
Comment (achen2401) [reply portal/static/js/main.js]: @ivan-c ivan, the fields are hidden by css in the stylesheet (e.g. study id is only visible in eproms, site id is only visible in truenthUSA) - that is why I am checking the visibility of each before updating it.

File: portal/static/js/main.js (hunk 2 above)
Comment (achen2401) [reply portal/static/js/main.js]: added, ivan!

**Pull request merged**